### PR TITLE
request actions: change redux to context

### PR DESCRIPTION
**Pull request by kpsherva**
* ensures easy re-usability of the actions components

:heart: Thank you for your contribution!

### Description

Please describe briefly your pull request.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Review by ntarocco — COMMENTED** [review]
LGTM! Some minor comments...

File: invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js [hunk 1]
```diff
@@ -4,71 +4,74 @@
...
-    const { setActionModalOpen, modalId } = this.props;
-    setActionModalOpen(isOpen, modalId);
+  static contextType = RequestActionContext;
+
+  componentDidMount() {
+    this.subscribeToContext();
+  }
+
+  subscribeToContext = () => {
+    const { modalId } = this.props;
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js:25]: its an action ID

File: invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionController.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  constructor(props) {
+    super(props);
+    const { request, requestApi } = props;
+    this.linkExtractor = new RequestLinksExtractor(request);
+    if (requestApi === undefined) {
+      this.requestApi = new InvenioRequestsAPI(this.linkExtractor);
+    } else {
+      this.requestApi = requestApi;
+    }
+    console.log(this.linkExtractor)
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionController.js:23]: thanks!

File: invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js [hunk 3]
```diff
@@ -4,71 +4,74 @@
...
+    let { modalOpen } = this.context;
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js:25]: You actually don't. `const` should block you to recreate the `obj`, so you cannot do `modalOpen = {}`, but you can assign keys/values to it. :)

**Review by ntarocco — APPROVED** [review]

File: invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js:25]: true!

**Pull request merged**